test/hotspot/jtreg/compiler/loopopts/superword/TestMultiversionRemoveUselessSlowLoop.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.loopopts.superword;
+
+/*
+ * @test
+ * @bug 8350756
+ * @summary Test case where the multiversion fast_loop disappears, and we should
+ *          constant fold the multiversion_if, to remove the slow_loop.
+ * @library /test/lib /
+ * @run driver compiler.loopopts.superword.TestMultiversionRemoveUselessSlowLoop
+ * @run main/othervm -XX:CompileCommand=compileonly,*Test*::test
+ *                   -XX:-TieredCompilation -Xcomp -XX:PerMethodTrapLimit=0
+ *                   compiler.loopopts.superword.TestMultiversionRemoveUselessSlowLoop
+ */
+
+public class TestMultiversionRemoveUselessSlowLoop {
```

Comment:
Is it also possible to add an IR test where we can check that a `OpaqueMultiversioning` node is first in the graph and then removed? Maybe we can check that in loop opts phase n, `COUNTED_LOOP_MAIN` and `OpaqueMultiversioning` is present. Then in loop opts phase n + 1, `COUNTED_LOOP_MAIN` is removed and in n + 2, `OpaqueMultiversioning` is removed as well. But I'm not sure if this is reliable enough when some loop opts change - though could easily be fixed or dropped again if the test fails at some point.

Reply:
I thought about it. I thought it would be quite hard to get a good test for this. Especially because it depends on the flags. But maybe I can create a test that has just the right flags, make it flagless, somehow engineer that the fast main loop decays in some specific loop-opts phase, and then in the next phase things must be cleaned up.

The issue is: in the current example the slow_loop already is removed anyway. But it would be nice to have a case where after it all the fast pre loop would still be around, or the fast post loop... hmm.

Maybe I can do it with fully unrolling the fast main loop, and the slow loop does not disappear because of it. Really tricky. Do you have a good idea @chhagedorn ? How much time would you like me to invest in this?

Reply:
@chhagedorn I created an IR test, let me know if it looks good enough for you :)

Reply:
Nice test! Thanks for taking the extra effort. I only have a small suggestion for it.